Configuring Docker Support along with Ollama Container

sounishnath003 opened the pull request:
The pull request I submitted includes the following changes: 
- I configured the `app.dockerfile` for the Svelte client.
- I configured the `devika.dockerfile` for the Devika backend engine.
- I also added `docker-compose` support to make it easy to run the entire infrastructure locally or anywhere else.

This pull request addresses issues #81 and #82, which are related to Docker configuration.


@github/sounishnath0030

BrianAndersen78 commented:
I had to change the entrypoint in app.dockerfile: 

ENTRYPOINT [ "npm", "run", "dev", "--", "--host" ]

Beside that it works as expected :)

sounishnath003 commented:
Kudos and thanks, @BrianAndersen78 for the mention. 

As perfectly identified, I am about to commit the same to my branch. as I updated the package.json in the `client-svelte-app`. I modified the `package.json` in ui which was forgotten during the git push. Now the concerned issue has been resolved by adding the additional flags to `ENTRYPOINT [ "npm", "run", "dev", "--", "--host" ]` in app.dockerfile

The same has been updated in the recent commits.

thanks and regards,
@github/sounishnath003

sounishnath003 commented:
Dear stitionai/devika,

I would like to express my gratitude for bringing to my attention that
@ezzcodeezzlife has approved the aforementioned pull request. Please be
informed that I have taken note of the update and appreciate the
notification.

Thank you for your continued support and cooperation in this matter.

Sincerely,

sounish

On Sun, Mar 24, 2024 at 9:24 PM fabi.s ***@***.***> wrote:

> ***@***.**** approved this pull request.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/stitionai/devika/pull/101#pullrequestreview-1956588732>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJTHRUJCCPU4CC66JQYVNKTYZ3ZKXAVCNFSM6AAAAABFFMYGHKVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTSNJWGU4DQNZTGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Will always follow 😊.
Thanks and Regards,

Sounish Nath
Software Engineer
Mobile: +918017204855
Portfolio <https://sounishnath.netlify.app/> | Github
<https://www.github.com/sounishnath003> | LinkedIn
<https://www.linkedin.com/in/sounishnath/> | Leetcode
<https://leetcode.com/sounish/>

schueszy commented:
@sounishnath003 
Thanks again for your Docker implementation.
But I still have a small comment and that is that the UI cannot connect to the API if the web interface was started on another computer.

guilherme-argentino commented:
I made it work by doing some adjustments and I even started to do some work, but when it starts to navigate in the browser, it simply dies. it doesn't do anything

sounishnath003 commented:
have you tried applying latest patch made on `app.dockerfile` .. cause the issue was reported earlier, and it was fixed in the following comment: https://github.com/stitionai/devika/pull/101#issuecomment-2016770358 .
And the app is up and running as usual.

schueszy commented:
@guilherme-argentino 
I got the same issure. add
```
RUN playwright install --with-deps
```
in the line below:
```
RUN $HOME/.cargo/bin/uv pip install -r requirements.txt 
```
in the devika.dockerfile. This should fix the issue.

schueszy commented:
@sounishnath003 
Can we add 
```
RUN UV_HTTP_TIMEOUT=100000 $HOME/.cargo/bin/uv pip install -r requirements.txt 
```
instead of
```
RUN $HOME/.cargo/bin/uv pip install -r requirements.txt 
```
Because the default value of UV_HTTP_TIMEOUT=300 is not sufficient for people who have a weak Internet connection.
see: https://github.com/astral-sh/uv/issues/2572

sounishnath003 commented:
@schueszy  - 

Great to see you all again. Prescriptive changes have been applied. Everything's fine now.
- Added playwright dependency support `RUN playwright install --with-deps` in `devika.dockerfile`
- Updated uv-http-timeout limit extending to `100000`.

Enjoy and Keep rocking...

schueszy commented:
@sounishnath003 
Ok i try it again now i get following error:
```
238.9 E: Failed to fetch http://deb.debian.org/debian/pool/main/libg/libglvnd/libopengl0_1.6.0-1_amd64.deb  Unable to connect to deb.debian.org:http: [IP: 146.75.122.132 80]
238.9 E: Unable to fetch some archives, maybe run apt-get update or try with --fix-missing?
238.9 Failed to install browsers
238.9 Error: Installation process exited with code: 100
------
failed to solve: process "/bin/sh -c playwright install --with-deps" did not complete successfully: exit code: 1
```
It is possible to add:
```
RUN apt-get update
```
before
```
RUN playwright install --with-deps
```

schueszy commented:
@sounishnath003 
Can you check my PR on:
https://github.com/sounishnath003/devika/pull/1/files

schueszy commented:
@vital121 created also a docker implementation #126

Israel-Laguan commented on ui/package-lock.json:
This one is not needed, please remove this change

sounishnath003 replied:
done

sounishnath003 replied:
done. removed package.json file

sounishnath003 commented:
To all -
- added `Makefile` support for easily run `devika` services in docker environment
- helps you to develop system in a `dev.container` environment as well

**Makefile command**
- `make setup` - to set requirements in your local environment
- `make compose-up` to run the entire system with services in Docker environment
- `make compose-destroy` - to purge out and delete dangling / prune volume and image resources

<img width="605" alt="image" src="https://github.com/stitionai/devika/assets/40270033/47fbc8c1-e9a0-4af5-960d-bb923f629150">

mufeedvh commented:
Thank you so much @sounishnath003 and everyone! Merging this now! ❤️🙌
